### PR TITLE
Calendar view

### PR DESCRIPTION
**Pull request by caromurp**
Uses react-native-big-calendar library to display schedule events in calendar format.  Defaults to pod schedule.  In addition, events are clickable but you cannot edit/delete events that you do not own.

### CODE REVIEW

File: mobile/src/events/CreateModifyEvent.tsx [hunk 1]
```diff
@@ -212,12 +214,14 @@ const CreateModifyEvent: React.FC<CreateModifyEventProps> = ({
...
-            />
-            {!!event && (
+            {!!event && user.id == event.ownerId && (
+                <Button
+                  onPress={handleSubmit}
+                  title="Save"
+                  style={[{ margin: 0 }, !isValid && sharedStyles.disabledButton]}
+                />
+            )}
+            {!!event && user.id == event.ownerId &&(
```
Comment (chloeprezelski) [inline mobile/src/events/CreateModifyEvent.tsx]: space plz

**Review by zusoomro — APPROVED** [review]
Looks good to me!

File: mobile/src/events/Schedule.tsx [hunk 2]
```diff
@@ -50,13 +62,24 @@ const ScheduleHomePage: React.FC<{}> = ({ navigation }) => {
 const Schedule: React.FC<{}> = ({ navigation }) => {
   const events = useSelector((state: RootState) => state.events.events);
   const [map, setMap] = useState([]);
-  const [pod, setPod] = useState<Pod>();
+  // const [pod, setPod] = useState<Pod>();
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: remove commented out line

File: mobile/src/events/Schedule.tsx [hunk 3]
```diff
@@ -91,30 +114,63 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
     } else {
       setLoading(true);
-      fetchPodEvents(pod.id).then((res) => {
-        // Line below should be dispatch(setEvents(res)
-        // NOT setEvents(res[0]) for the schedule toggle to work
-        dispatch(setEvents(res));
-        setLoading(false);
+      fetchUserPod().then((res) => {
+        if (res) {
+          fetchPodEvents(res.id).then((res2) => {
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: can we use different names here besides res and res2. It's really confusing to the reader to know what these are and the code would be more readable if we knew what res and res2 were by providing meaningful variable names

File: mobile/src/events/Schedule.tsx [hunk 4]
```diff
@@ -91,30 +114,63 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
       });
     }
   }, [isToggledToUser, pod]);
 
+  // If events change, process them into calendar events
+  React.useEffect(() => {
+
+    const PromiseArray = events.map(async (event) => {
+      const email = await fetchUserEmail(event.ownerId);
+      console.log("event in forEach loop:" ,event)
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: remove console.log statement

File: mobile/src/events/Schedule.tsx (hunk 2 above)
Comment (allysmith12) [inline mobile/src/events/Schedule.tsx]: delete this

File: mobile/src/events/Schedule.tsx [hunk 5]
```diff
@@ -91,30 +114,63 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
+        start: event.start_time,
+        end: event.end_time,
+        id: event.id,
+        wigoEvent: event,
+        children: <View><Text style={{color: "#FFFFFF", fontSize: 12}} numberOfLines={1}>{email.split("@")[0]}</Text></View>
+      }
+    })
+
+    Promise.all(PromiseArray).then((values) => {
+      console.log("processed events:", values)
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: is this needed?

File: mobile/src/events/Schedule.tsx [hunk 6]
```diff
@@ -91,30 +114,63 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
+
   const dd = String(today.getDate()).padStart(2, "0");
   const mm = String(today.getMonth() + 1).padStart(2, "0");
   const yyyy = today.getFullYear();
   const todayString = mm + "/" + dd + "/" + yyyy;
 
   return (
     <View style={{ flex: 1 }}>
-      <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
+      <View contentContainerStyle={{ flexGrow: 1 }}>
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx:169]: if there were a lot of events in one day, would this scroll off the page? Should it still be a ScrollView. IDK just asking

File: mobile/src/events/Schedule.tsx [hunk 7]
```diff
@@ -130,15 +186,27 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
-              <EventInSchedule
-                event={event}
-                showName={!isToggledToUser}
-                navigation={navigation}
-                key={event.id}
-                avatar={isToggledToUser ? user.avatar : map[event.ownerId]}
-              />
-            ))}
+            {processedEvents && processedEvents.length != 0 &&
+
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: remove empty line. there's some formatting issues in this PR that could prob be fixed with yarn run formatAll :)

File: mobile/src/events/Schedule.tsx [hunk 8]
```diff
@@ -130,15 +186,27 @@ const Schedule: React.FC<{}> = ({ navigation }) => {
...
+            (<Calendar events={processedEvents!} mode="day" scrollOffsetMinutes={480}
+              height={Dimensions.get('window').height - 130} ampm={true} onPressEvent={(event: CalendarEvent) => { 
+                console.log(`Clicked on event with id: ${event.id}`); 
+                navigation.navigate("ModifyEvent", { event: events.find((eventObject: Event) => eventObject.id == event.id), pod })
+              }} 
+              swipeEnabled={true}
+              eventCellStyle={(event: BaseEvent) => {
+                const fixedEvent = event as CalendarEvent;
+                const userId = fixedEvent.id
+                console.log(podMemberIdToColor[fixedEvent.wigoEvent.ownerId]);
```
Comment (chloeprezelski) [inline mobile/src/events/Schedule.tsx]: remove / are they needed?

**Pull request merged**